### PR TITLE
Live dashboard checkin

### PR DESCRIPTION
**Pull request by karbmk**
Automatically checks in the volunteer if he/she is checked when the event is live.
Files changed:
1. live-dashboard.html
2. views.py

### CODE REVIEW

**Comment by karbmk** [comment]
#228

**Comment by nickolashe** [comment]
lets resolve the conflict pls

**Comment by nickolashe** [comment]
@karbmk @tsiology @dannypernik I am seeing volunteers added to the dashboard checked in on the backend, but the checkbox stays gray. Can you confirm?

**Comment by nickolashe** [comment]
bump

**Comment by karbmk** [comment]
Hi @nickolashe , I verified this. It shows blue tick only when its happening now. Else it will just be added to the usereventregistration.

**Comment by nickolashe** [comment]
@karbmk are you saying that it only checks in the volunteer when the event is happening? or only colors the button so? i think i saw that it checked in a volunteer on my request when the event was not live. can we double check?

**Comment by karbmk** [comment]
Sure

**Comment by karbmk** [comment]
Hi @nickolashe , I checked it. What it basically is doing is add the user in usereventregistration whether it is a live event or an event in the future. But, Will check-in(add in usertimelog) only if it's a live event. If it's an event in the future then it won't create a usertimelog for that user.

**Comment by nickolashe** [comment]
ok, thanks - i'll check again

**Comment by nickolashe** [comment]
@karbmk got it, i think this is correct.

**Comment by nickolashe** [comment]
i think what i meant originally is volunteers can still be checked in by pressing the checkbox for events far in future. but i think this should be a separate issue.

**Comment by nickolashe** [comment]
@karbmk as the last step, do you think you can include changes addressing https://github.com/opencurrents/opencurrents/issues/112 in this PR as well?

**Comment by karbmk** [comment]
Hi Nicko, Sure. If I understand it properly, the opt-in should always be done explicitly. the live-dashboard auto-checkin should be removed. Is that correct?

**Comment by nickolashe** [comment]
@karbmk What do you mean by auto-checkin?

**Comment by dannypernik** [comment]
screen shot 2017-08-09 at 5 05 44 am

@karbmk for clarity, the checkbox mentioned in #112 is the "Invite volunteer to openCurrents" checkbox shown above. Basically after the admin clicks "Add volunteer", the checkbox should reset. Currently, it stays checked.

**Comment by karbmk** [comment]
Oh got it now. I will take a look.

File: openCurrents/views.py [hunk 1]
```diff
@@ -1100,15 +1100,28 @@ def event_register_live(request, eventid):
...
-        user=user,
-        event=event,
-        is_confirmed=True
-    )
-    user_event_registration.save()
-    logger.info('User %s registered for event %s', user.username, event.id)
-
-    return HttpResponse({'userid': userid, 'eventid': eventid}, status=201)
+    user_events = UserEventRegistration.objects.values('user__id','event__id').filter(user__id = userid)
+    user_event_ids = [d for d in user_events if int(userid) == d['user__id'] and int(eventid) == d['event__id']]
```
Comment (nickolashe) [inline openCurrents/views.py]: @karbmk why do we need to build this list when we could add another filter the line above to based on matching eventid?

File: openCurrents/views.py [hunk 2]
```diff
@@ -1100,15 +1100,28 @@ def event_register_live(request, eventid):
...
+        )
+        user_event_registration.save()
+        logger.info('User %s registered for event %s', user.username, event.id)
+    else:
+        logger.info('User %s already registered for event %s', user.username, event.id)
+        return HttpResponse(status=400)
+    tz = event.project.org.timezone
+    event_ds = event.datetime_start.astimezone(pytz.timezone(tz)).time()
+    event_de = event.datetime_end.astimezone(pytz.timezone(tz)).time()
+    d_now = datetime.utcnow().replace(tzinfo=pytz.utc).astimezone(pytz.timezone(tz))
```
Comment (nickolashe) [inline openCurrents/views.py]: @karbmk this does not have to be converted in org's timezone. tz-aware datetimes can be compared directly even if they are in different timezones.

**Comment by nickolashe** [comment]
@karbmk looks good, just a couple of changes based on the comments above

File: openCurrents/views.py (hunk 2 above)
Comment (karbmk) [inline openCurrents/views.py]: Oh I did'nt know that. Thanks i will change that.

File: openCurrents/views.py (hunk 1 above)
Comment (karbmk) [inline openCurrents/views.py]: I just need the numbers inside the list but filter is building a list of dictionaries with event__id as key for each element which is sort of not what i wanted.

Comment (nickolashe) [inline openCurrents/views.py]: @karbmk I am not seeing where you are using those numbers though. I am seeing all you are checking for is whether the list is emtpy or not.

Comment (karbmk) [inline openCurrents/views.py]: Hmm, You're right. I thought it was being passed in the URL or something. changing it.

**Comment by nickolashe** [comment]
@karbmk looks good

**Pull request merged**